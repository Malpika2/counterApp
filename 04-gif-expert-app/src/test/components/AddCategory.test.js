import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en el componente AddCategory', () => {

    const setCategory = jest.fn();
    let wrapper = shallow( <AddCategory setCategory={setCategory} />);

    beforeEach( () => {
        jest.clearAllMocks(); //Limpia simulaciones
        wrapper = shallow( <AddCategory setCategory={setCategory} />);
    });

    test('Debe Mostrarse Correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', { target:{ value } } );
        
    });

    test('No debe hacer el submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){} } );

        expect( setCategory ).not.toHaveBeenCalled();
    })
    
    
    
});