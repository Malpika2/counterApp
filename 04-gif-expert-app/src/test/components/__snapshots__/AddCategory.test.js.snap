// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe Mostrarse Correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add category
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
